holmes/checks/checks.py

```
@@ -0,0 +1,673 @@
+            filtered_checks = [
+                c for c in filtered_checks if any(tag in c.tags for tag in tag_filter)
+            ]
+        return filtered_checks
+
+    def _validate_alert_destinations(self, filtered_checks, destinations_config):
+        """Warn if any alert-mode checks have no destinations configured."""
+        alert_checks_with_no_destinations = [
+            c.name
+            for c in filtered_checks
+            if c.mode == CheckMode.ALERT
+            and not c.destinations
```

Ruff (0.14.13)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)